fifo_queue.go

```diff
@@ -180,6 +180,18 @@ func (st *FIFO) Remove(index int) error {
 	return nil
 }
 
+// GetAll returns the entire list of elements from the queue
+func (st *FIFO) GetAll() (interface{}, error) {
```

Comment:
This method should have some parameters available to paginate the response. The parameters used in this function do not have to be the same as the pagination (`page` & `page_size`) ones. They can be if you want, I'll leave that up to you. If the parameters are not the same, the service/repository handling this data structure will be in charge of translating between the API parameters and method parameters.